cupyx/scipy/sparse/_data.py

```diff
@@ -238,7 +238,7 @@ def _arg_min_or_max(self, axis, out, op, compare):
                 if compare(m, zero):
                     return mat.row[am] * mat.shape[1] + mat.col[am]
                 else:
-                    size = cupy.prod(mat.shape)
+                    size = cupy.prod(cupy.asarray(mat.shape))
```

Comment:
How about the following simplification?
```suggestion
                    size = mat.size
```

Reply:
I think `size` for a sparse matrix refers to the number of non-zero entries, which is different from the total elements that would be on its dense version

Reply:
I am not sure what @asi1024 had in mind, but I think the variable `size` is needed on CPU for conditional branches, so it does not make sense to compute and keep it on GPU. By the time we need to do `if size == mat.nnz` it's moved back to CPU again. (Though apparently the mistake was done earlier, not in this PR 😅)

Reply:
Yes, I want to keep `size` in CPU memory. How about using `numpy.prod` or `math.prod` instead?